lib/streamlit/cli.py

```diff
@@ -147,6 +147,7 @@ def configurator_options(func):
             parsed_parameter["param"],
             help=parsed_parameter["description"],
             type=parsed_parameter["type"],
+            envvar='STREAMLIT_CONFIG_{}'.format(parsed_parameter["param"].upper()),
```

Comment:
Make the suffix use snake case too. We have a function called `to_snake_case()` in `case_converters.py` that should do the trick.

In which case this line will get really long and nasty, so you could pull this logic into the `_convert_config_option_to_click_option` function.

Even better if we could change line 145 to just
```
config_option = click.option(**parsed_parameter)
```

Reply:
Sure, I'll take a look at all these tomorrow!

Reply:
I've moved the logic into `_convert_config_option_to_click_option()`. But, I'm not so sure about applying `to_snake_case`.

For an option like `server.baseUrlPath`, we have:

`STREAMLIT_CONFIG_SERVER_BASE_URL_PATH` (with snakecase)
`STREAMLIT_CONFIG_SERVER_BASEURLPATH` (without snakecase)

I think the without snakecase version matches the underlying config name better. With the snakecase version, I wouldn't know if it's supposed to match a config named `server.base.url.path`, `server.baseUrl.path`, `serverBaseUrlPath`, etc.

Reply:
The plan is to have only a two-level hierarchy for config options, so there shouldn't be any ambiguity so I think `STREAMLIT_CONFIG_SERVER_BASE_URL_PATH` is better.

Reply:
The non-snakecase version has already been merged. But, here's how some other projects do environment variable to config mappings, for comparison. Overall, I feel that converting camelCase to straight uppercase without snake-casing is the better choice because it matches more closely with other apps.

**Click's Default Behaviour**

Turning on automatic environment variables in Click converts option names to uppercase, without snake casing.

`--server.baseUrlPath` == `{PREFIX}_SERVER_BASEURLPATH`

**Traefik**

[Traefik's configs](https://docs.traefik.io/v2.0/reference/static-configuration/file/) have camelCase names, and are mapped to [CLI options](https://docs.traefik.io/v2.0/reference/static-configuration/cli/) in lowercase. The corresponding [environment variables](https://docs.traefik.io/v2.0/reference/static-configuration/env/) are not snake-cased.

`[accessLog] filePath` == `--accesslog.filepath` == `TRAEFIK_ACCESSLOG_FILEPATH`

**Kafka (using Confluent Platform images)**

[Kafka's configs](https://kafka.apache.org/documentation/#brokerconfigs) are all lowercase, and tend to have each word separated by a period (as opposed to camelCasing or snake_casing). In the [Confluent Platform image for Kafka](https://hub.docker.com/r/confluentinc/cp-kafka/), the corresponding environment variables are not snake-cased.

`auto.create.topics.enable` == `KAFKA_AUTO_CREATE_TOPICS_ENABLE`

**Grafana**

[Grafana's configs](https://grafana.com/docs/installation/configuration/) are snake cased, which are preserved in [environment variables](https://grafana.com/docs/installation/configuration/#using-environment-variables).

`[server] http_port` == `GF_SERVER_HTTP_PORT`

Reply:
Ooops, I think I merged this too soon. No problem, we'll address it in another PR.